src/sql.rs

```diff
@@ -747,10 +750,117 @@ pub async fn housekeeping(context: &Context) -> Result<()> {
         .await
         .ok_or_log_msg(context, "failed to remove old MDNs");
 
+    let start_time = Instant::now();
+    let mut msg_cnt: usize = 0;
+    loop {
+        let select = |conn: &mut Connection| {
+            let mut msgs = Vec::new();
+            let start_time = Instant::now();
+            let mut stmt = conn.prepare(
+                "SELECT id, mime_headers FROM msgs WHERE mime_headers!='' AND mime_compressed=0",
```

Comment:
are we sure,, this does not load all `mime_headers` to memory at once?

surely, it should not, but after the recent experiences with https://github.com/deltachat/deltachat-core-rust/pull/4164 i am a bit sceptical ... maybe we should still add some `LIMIT` just to sleep better, also if things may change in the future?

Reply:
k, this was not changed by the recent changes wrt transaction, so already tested and probably okay.

@iequidoo i prefer to have some commit history, even if force-push is used, this makes sense - currently, there is no way to compare the current state to the one before @link2xt adapted the loop (i have a local backup by random, however :)

Reply:
k, i measure the memory usage on my database, this is on my M1 pro macos max. 25mb per second, this is probably quite okay, even for lower budget devices - however, these devices will probably be also slower and use less memory per second.

Reply:
> are we sure, this does not load all `mime_headers` to memory at once?
> 
> surely, it should not, but after the recent experiences with #4164 i am a bit sceptical ... maybe we should still add some `LIMIT` just to sleep better, also if things may change in the future?
> 
> even if that is okay - how much memory is used during the 1 s where all data are collected in memory?

But there's no good LIMIT value, we don't know the avg size of messages, so we should reduce the time limit instead if we see a huge memory usage. SQLite itself doesn't eat much memory because of SELECT wth no LIMIT, probably it has a buffer of some reasonably limited size, the question is how much does our `msgs` `Vec` eat. On my machine i don't see in the `top` `RES` > 50m, but better you check on your large db-s.

And btw, i don't see any performance improvement when increasing the time limit from the current 1s to 10s, so maybe even 1s is too much, but i think it's hardware-dependent, so i'd left it as is for now.